### PR TITLE
Added bxt_ch_set_angles

### LINKED ISSUE #147: bxt_ch_set_angles <pitch> <yaw>

### PR DESCRIPTION
**Pull request by chinese-soup**
Closes #147 

EDIT: Just noticed that there's some floating point error in setting the yaw, not sure why, but `bxt_ch_set_angles 10 10` ends up as:

image

Any idea how to fix that?

`hudSetViewAngles` is:
```c++
void hudSetViewAngles(float *va)

{
  (*g_engdstAddrs.SetViewAngles)(&va);
  cl.viewangles[0] = (vec_t)*va;
  cl.viewangles[1] = (vec_t)va[1];
  cl.viewangles[2] = (vec_t)va[2];
  return;
}
```

Also, should the cheat also include the roll as an argument, seeing as it's a cheat anyway (and roll is sometimes set - e.g. when player dies and deathcam starts)? What do you think, @SmileyAG?

### CODE REVIEW

**Comment by SmileyAG** [comment]
Well done! 

If you wish add <roll> argument — you could do that, but make next thing:

For example, if i'll miss <roll> argument, could you set it by default to 0?

Because as far as I know, the roll angle is zero in most cases, but I could be wrong.

**Comment by YaLTeR** [comment]
> there's some floating point error in setting the yaw

There's no error, it's anglemod. This is expected behavior.

File: BunnymodXT/modules/HwDLL.cpp [hunk 1]
```diff
@@ -1755,6 +1761,22 @@ struct HwDLL::Cmd_BXT_CH_Set_Origin_Offset
 	}
 };
 
+
```
Comment (YaLTeR) [inline BunnymodXT/modules/HwDLL.cpp:1764]: the usual
```suggestion
```

Comment (chinese-soup) [reply BunnymodXT/modules/HwDLL.cpp]: :see_no_evil:

**Comment by chinese-soup** [comment]
> > there's some floating point error in setting the yaw
> 
> There's no error, it's anglemod. This is expected behavior.

Alright, great.

> If you wish add "roll" argument — you could do that, but make next thing:
> 
> For example, if i'll miss "roll" argument, could you set it to 0 as like default?
> 
> Because as far as I know, the roll angle is zero in most cases.

Alright, done. :+1:

**Review by YaLTeR — APPROVED** [review]

**Comment by chinese-soup** [comment]
Drafting to include Windows support, I guess.

**Comment by chinese-soup** [comment]
@SmileyAG Works on Windows now as well, dunno since when since I've always force pushed like the smart person I am. ;)

**Comment by YaLTeR** [comment]
> @YaLTeR Should I just remove the hudSetViewAngles hook that only works on Linux and use the ClientDLL exported engine functions for both Linux and Windows?

Makes sense. Also

image

**Comment by chinese-soup** [comment]
Should be good to go now.

**Comment by YaLTeR** [comment]
Could you also add a check for pEngfuncs being NULL? To do nothing in that case.

**Comment by chinese-soup** [comment]
Done.

**Comment by YaLTeR** [comment]
Thanks

**Pull request merged**